tests/test_wires.py

```diff
@@ -377,3 +377,171 @@ def test_wires_pytree(self, source):
         wires2 = tree_unflatten(tree, wires_flat)
         assert isinstance(wires2, Wires), f"{wires2} is not Wires"
         assert wires == wires2, f"{wires} != {wires2}"
+
+    @pytest.mark.parametrize(
+        "wire_a, wire_b, expected",
+        [
+            (Wires([0, 1]), Wires([2, 3]), Wires([0, 1, 2, 3])),
```

Comment:
Just putting a note here that when you add functionality for allowing `other` to be a `list`, add another test like this for each set combination method:

```suggestion
        [
            ([0, 3, 4], Wires([2, 3]), Wires([0, 2, 3, 4])), 
            (Wires([0, 1]), Wires([2, 3]), Wires([0, 1, 2, 3])),
```